### PR TITLE
fix(apple): Disable `enableNativeCrashHandling` and `enableAutoPerformanceTracing`

### PR DESCRIPTION
**Pull request by mwerder**
fix cast to BOOL for sentry init options 'enableNativeCrashHandling' and 'enableAutoPerformanceTracing'

## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Environment:
XCode: 14.2
sentry/react-native: ^5.2.0
platform: MacOS Catalyst & iOS

I integrated sentry into our react-native catalyst app for MacOS.
While building I ran into following errors:

`"cast from pointer to smaller type 'bool' (aka 'signed char') loses information"`
on following lines inside _RNSentry.mm_:

ln:111 `BOOL enableNativeCrashHandling = (BOOL)[mutableOptions valueForKey:@"enableNativeCrashHandling"];`

ln:122`BOOL enableAutoPerformanceTracing = (BOOL)[mutableOptions valueForKey:@"enableAutoPerformanceTracing"];`

I'm not familiar with objective-c and its types. As the type BOOL behaves differently on 32 and 64-bit machines, I thought that was the problem. But both, iOS and MacOS are 64-bit systems.
So I cannot figure out why it works on iOS but it throws an error on MacOS.

What I found out after testing though, is that with the existing cast, the options _enableNativeCrashHandling_ and _enableAutoPerformanceTracing_ are always true even if I set them to false inside the Sentry.init() options (test cases 5 and 6 below failed with the existing code using the (BOOL) cast).

By debugging I can confirm that following if statement is never entered when setting _enableNativeCrashHandling_ to false:

```
BOOL enableNativeCrashHandling = (BOOL)[mutableOptions valueForKey:@"enableNativeCrashHandling"];

if (!enableNativeCrashHandling) {

  NSMutableArray *integrations = sentryOptions.integrations.mutableCopy;
  [integrations removeObject:@"SentryCrashIntegration"];
  sentryOptions.integrations = integrations;
}
```

## :bulb: Motivation and Context
Fixes catalyst build and allows options _enableNativeCrashHandling_ & _enableAutoPerformanceTracing_ to be disabled on iOS

The solution is based on following answer on [Stack Overflow](https://stackoverflow.com/a/34600806)

## :green_heart: How did you test it?
I ran tests locally while running the app on iOS and macOS. Tested all options using console.logs to verify the options are set correctly.
As mentioned above, using the existing code I always get true. So test case 5 and 6 are failing.
This would mean that it is not possible to disable AutoPerformanceTracing and NativeCrashHandling.
I tried to look into the test cases but honestly, I'm not sure where to add them as the options seems to be mocked.
Would this need an e2e test to verify the iOS code?

| # | Sentry.init options | expected Result | actual Result iOS | actual Result MacOS | Pass/Fail |
| :---         |     :---:      |          :---: |         :---: |          :---: | :---: |
|1   | enableAutoPerformanceTracing: not set (default)     | 1    | 1    | 1    | ✅  |
|2   | enableNativeCrashHandling: not set (default)     | 1    | 1    | 1    | ✅  |
|3   | enableAutoPerformanceTracing: true     | 1    | 1    | 1    | ✅  |
|4   | enableNativeCrashHandling: true     | 1    | 1    | 1    | ✅  |
|5   | enableAutoPerformanceTracing: false    | 0    | 0    | 0    | ✅  |
|6   | enableNativeCrashHandling: false    | 0    | 0    | 0    | ✅  |

closes: https://github.com/getsentry/sentry-react-native/issues/2944

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
Maybe add test cases for options.

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
Thank you for the PR, 
the changes look good, also thank you for the detailed description, that is awesome.

We have [a few tests for the method `createOptionsWithDictionary`](https://github.com/getsentry/sentry-react-native/blob/d197b5c9ca10ac3f729d1b3ef805328665aa0895/RNSentryTester/RNSentryTesterTests/RNSentry+initNativeSdk.mm#L33) would you be to add few to the changes you made to avoid future regression. (enable/disable `enableNativeCrashHandling` and check if the the integrations are present or not, enable/disable `enableAutoPerformanceTracing` and check if the private flags are set correctly)

File: RNSentryTester/RNSentryTesterTests/RNSentry+initNativeSdk.mm [hunk 1]
```diff
@@ -29,6 +29,60 @@ - (void)testCreateOptionsWithDictionaryRemovesPerformanceProperties
...
+        @"enableNativeCrashHandling": @NO,
+        @"enableAutoPerformanceTracing": @NO,
+
+    };
+    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
+    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
+    XCTAssertNil(error, @"Should not pass no error");
+    XCTAssertEqual([actualOptions.integrations containsObject:@"SentryCrashIntegration"], false, @"Did not disable native crash handling");
+    XCTAssertEqual(actualOptions.enableAutoPerformanceTracing, false, @"Did not disable Auto Performance Tracing");
+}
```
Comment (krystofwoldrich) [inline RNSentryTester/RNSentryTesterTests/RNSentry+initNativeSdk.mm:85]: I like the tests, I would just split the crash handling and performance options as they are independent.

```suggestion
- (void)testCreateOptionsWithDictionaryNativeCrashHandlingDefault
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual([actualOptions.integrations containsObject:@"SentryCrashIntegration"], true, @"Did not set native crash handling");
}

- (void)testCreateOptionsWithDictionaryPerformanceTrackingDefault
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual(actualOptions.enableAutoPerformanceTracing, true, @"Did not set Auto Performance Tracing");
}

- (void)testCreateOptionsWithDictionaryNativeCrashHandlingEnabled
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
        @"enableNativeCrashHandling": @YES,
    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual([actualOptions.integrations containsObject:@"SentryCrashIntegration"], true, @"Did not set native crash handling");
}

- (void)testCreateOptionsWithDictionaryPerformanceTrackingEnabled
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
        @"enableAutoPerformanceTracing": @YES,

    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual(actualOptions.enableAutoPerformanceTracing, true, @"Did not set Auto Performance Tracing");
}

- (void)testCreateOptionsWithDictionaryNativeCrashHandlingDisabled
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
        @"enableNativeCrashHandling": @NO,

    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual([actualOptions.integrations containsObject:@"SentryCrashIntegration"], false, @"Did not disable native crash handling");
}

- (void)testCreateOptionsWithDictionaryPerformanceTrackingDisabled
{
    RNSentry * rnSentry = [[RNSentry alloc] init];
    NSError* error = nil;

    NSDictionary *_Nonnull mockedReactNativeDictionary = @{
        @"dsn": @"https://abcd@efgh.ingest.sentry.io/123456",
        @"enableAutoPerformanceTracing": @NO,

    };
    SentryOptions* actualOptions = [rnSentry createOptionsWithDictionary:mockedReactNativeDictionary error:&error];
    XCTAssertNotNil(actualOptions, @"Did not create sentry options");
    XCTAssertNil(error, @"Should not pass no error");
    XCTAssertEqual(actualOptions.enableAutoPerformanceTracing, false, @"Did not disable Auto Performance Tracing");
}
```

**Review by krystofwoldrich — APPROVED** [review]
Looks good. Thank you @mwerder and @birdofpreyru 🚀

**Comment by mwerder** [comment]
Glad I could contribute. Thanks for your help @krystofwoldrich 🚀

**Pull request merged**